src/game_inventory.cpp

```diff
@@ -2296,106 +2295,108 @@ drop_locations game_menus::inv::smoke_food( Character &you, units::volume total_
-
-                ui.position( point::zero, point( half_width * 2, height ) );
-            } );
-            ui.mark_resize();
-            ui.on_redraw( [&]( const ui_adaptor & ) {
-                if( !confirm_message.empty() ) {
-                    draw_border( wnd_message );
-                    nc_color col = c_white;
-                    print_colored_text(
-                        wnd_message, point( 3, 1 ), col, col,
-                        confirm_message + " " +
-                        ctxt.describe_key_and_name( "CONFIRM" ) + " " +
-                        ctxt.describe_key_and_name( "QUIT" ) );
-                    wnoutrefresh( wnd_message );
-                }
+game_menus::inv::compare_item_menu::compare_item_menu( const item &first, const item &second,
+        const std::string &confirm_message ) :
+    cataimgui::window( "compare", ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove |
+                       ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoNavInputs ),
+    first( first ),
+    second( second ),
+    confirm_message( confirm_message )
+{
+    ctxt.register_action( "HELP_KEYBINDINGS" );
+    if( !confirm_message.empty() ) {
+        ctxt.register_action( "CONFIRM" );
+    }
+    ctxt.register_action( "QUIT" );
+    ctxt.register_action( "UP" );
+    ctxt.register_action( "DOWN" );
+    ctxt.register_action( "PAGE_UP" );
+    ctxt.register_action( "PAGE_DOWN" );
+    ctxt.set_timeout( 10 );
+
+    // todo: regen info when toggling language?
+    first.info( true, first_info );
+    second.info( true, second_info );
+}
 
-                draw_item_info( wnd_first, item_info_first );
-                draw_item_info( wnd_second, item_info_second );
-            } );
-            lang_version = detail::get_current_language_version();
-            first_execution = false;
+static void draw_column( const std::string &label, const ImVec2 &size, const item &it,
+                         std::vector<iteminfo> &first, std::vector<iteminfo> &second, cataimgui::scroll &s )
+{
+    if( ImGui::BeginChild( label.c_str(), size ) ) {
+        cataimgui::TextColoredParagraph( c_light_gray, it.tname() );
+        ImGui::NewLine();
+        ImGui::NewLine();
+        if( it.tname().find( it.type_name() ) == std::string::npos ) {
+            cataimgui::TextColoredParagraph( c_light_gray, it.type_name() );
+            ImGui::NewLine();
         }
+        cataimgui::set_scroll( s );
+        display_item_info( first, second );
+    }
+    ImGui::EndChild();
+}
 
+void game_menus::inv::compare_item_menu::draw_controls()
+{
+    float half_width = ImGui::GetContentRegionAvail().x / 2;
+    float height = ImGui::GetContentRegionAvail().y;
+    ImGuiStyle &style = ImGui::GetStyle();
+    float spacing_x = style.ItemSpacing.x;
+    float top_y = ImGui::GetCursorPosY();
+    float confirm_height = confirm_message.empty() ? 0.f : style.ItemSpacing.y +
+                           style.FramePadding.y * 2 + ImGui::GetFontSize();
```

Comment:
I think `ImGui::GetFrameHeightWithSpacing()` does exactly what you want here.

Reply:
Oh, yes, thanks, I was looking for that within style, that's why I didn't find it.